python/desc/imsim/imSim.py

```diff
@@ -830,14 +831,17 @@ def add_cosmic_rays(gs_interpreter, phot_params):
 
     # Retrieve the visit number for the random seeds.
     visit = gs_interpreter.obs_metadata.OpsimMetaData['obshistID']
+    band = gs_interpreter.obs_metadata.bandpass
 
     exptime = phot_params.nexp*phot_params.exptime
     for name, image in gs_interpreter.detectorImages.items():
         imarr = copy.deepcopy(image.array)
         # Set the random number seed for painting the CRs.
-        crs.set_seed(CosmicRays.generate_seed(visit, name))
+        cr_seed = CosmicRays.generate_seed(visit, name)
+        crs.set_seed(cr_seed)
         gs_interpreter.detectorImages[name] = \
             galsim.Image(crs.paint(imarr, exptime=exptime), wcs=image.wcs)
+        image.wcs.fitsHeader.set('CR_SEED', cr_seed)
```

Comment:
Oops sorry missed this file... Adding as a separate comment.

So one thing I am a bit confused by.  In the seeds above for noise etc they went into the amp file (which makes sense).

By using wcs.fitsHeader.set does that mean it will go into the eimage + amp?

I'm also wondering about RATEL, DECTEL, ROTANGLE above which (I think) only goes to the amp file.  Should those go to both?

Reply:
OK sorry... I remember/see how this works now. It is grabbing RATEL etc from the header in the GSobject but there is a way to grab it from the opsim database if the info is missing in the old files.